tools/monorepo-utils/src/workflow-profiler/commands/profile/index.ts

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import { Command } from '@commander-js/extra-typings';
+
+/**
+ * Internal dependencies
+ */
+import {
+	getWorkflowRunData,
+	getWorkflowData,
+	getRunJobData,
+	getCompiledJobData,
+} from '../../lib/data';
+import {
+	logWorkflowRunResults,
+	logJobResults,
+	logStepResults,
+} from '../../lib/log';
+import { Logger } from '../../../core/logger';
+
+const program = new Command( 'profile' )
+	.description( 'Profile GitHub workflows' )
+	.argument( '<start>', 'Start date in YYYY-MM-DD format' )
+	.argument( '<end>', 'End date in YYYY-MM-DD format' )
+	.argument(
+		'<id>',
```

Comment:
Having to use `list` and then `profile` with an ID feels unnecessarily granular. How would you feel about using the workflow file instead and pulling the ID from that ourselves? Is that possible?

Reply:
This approach would also make the `list` command unnecessary. Personally, I find the output of that a little hard to follow since it's hard to understand what workflows _really_ do what. The specific events that a workflow runs on give more context about the workflow itself.

Reply:
> The specific events that a workflow runs on give more context about the workflow itself.

Cool idea. I'd need to parse the file itself to get that info.